mlflow/sklearn/__init__.py

```diff
@@ -884,12 +917,72 @@ def patched_fit(original, self, *args, **kwargs):
                           for autologging (e.g., specify "fit" in order to indicate that
                           `sklearn.linear_model.LogisticRegression.fit()` is being patched)
         """
+        global _autolog_training_status
         with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
             if t.should_log():
-                return fit_mlflow(original, self, *args, **kwargs)
+                _autolog_training_status.register_model(self, mlflow.active_run().info.run_id)
+                try:
+                    _autolog_training_status.in_fit_call_scope = True
+                    result = fit_mlflow(original, self, *args, **kwargs)
+                finally:
+                    _autolog_training_status.in_fit_call_scope = False
+                return result
             else:
                 return original(self, *args, **kwargs)
 
+    def patched_predict(original, self, *args, **kwargs):
+        from mlflow.utils import _inspect_original_var_name
+        global _autolog_training_status
+        with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
+            if t.should_log():
+                result = original(self, *args, **kwargs)
+                eval_dataset = args[0] if len(args) >= 1 else kwargs.get('X')
+                eval_dataset_name = _inspect_original_var_name(eval_dataset)
+                _autolog_training_status.register_predition_result(
+                    self, eval_dataset, result, eval_dataset_name)
+                return result
+            else:
+                return original(self, *args, **kwargs)
+
+    def log_eval_metric(metric_name, metric, metric_api_call_arg_list):
+        if not np.isscalar(metric):
+            # Skip logging metric which is non-scalar value
+            return
```

Comment:
I'd insert this right after `metric = original(*args, **kwargs)`.

Reply:
Can also check whether `metric` is numeric?

Reply:
not sure. i guess no

Reply:
I think we could investigate this using this approach: https://github.com/mlflow/mlflow/pull/4491#issuecomment-874433278